### ISSUE TITLE
`get` returns a decimal but `set` only takes an integer

### ISSUE THREAD

**Issue by sed-i** [canonical/multipass#2822]
**Describe the bug**
When I try to set the memory of an instance to a value which includes a decimal point, multipass rejects it. An integer works. This is inconsistent with `get`, which does return a decimal.

**To Reproduce**
```
$ multipass get local.charm-dev-2cpu-7g.memory                                                                                                                                      
7.0GiB

$ multipass set local.charm-dev-2cpu-7g.memory=8.0GiB                                                                                                                               
set failed: Invalid setting 'local.charm-dev-2cpu-7g.memory=8.0GiB': 8.0GiB is not a valid memory size - need a non-negative integer (in base 10) followed by the desired unit (e.g. 1234B, 42MiB)

$ multipass set local.charm-dev-2cpu-7g.memory=8GiB                                                                                                                                 

$
```

**Expected behavior**
`multipass set ....memory=8.0GiB` to work the same as `8GiB`. Or if integers is a design choice then `get` should also return an integer.

**Logs**
NTA.

**Additional info**
 - OS: Ubuntu 22.04.1 LTS
- multipass   1.10.1
- multipassd  1.10.1
- local.driver: qemu

**Additional context**
[K8s](https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/quantity/) takes both binary and decimal notation and converts decimals to "canonical form".

**Comment by ricab** [comment]
Hi @sed-i, this is already being tracked by #2677 and there is a PR in flight to fix it #2810.

Duplicate of #2677

**Comment by ricab** [comment]
Duplicate of #2677
